modules/store/src/reducer_creator.ts

```diff
@@ -229,15 +229,25 @@ export function createReducer<S, A extends Action = Action>(
   initialState: S,
   ...ons: On<S>[]
 ): ActionReducer<S, A> {
-  const map = new Map<string, ActionReducer<S, A>>();
+  const map = new Map<string, ActionReducer<S, A>[]>();
   for (let on of ons) {
     for (let type of on.types) {
-      map.set(type, on.reducer);
+      if (map.has(type)) {
```

Comment:
You could do the composition on the fly here instead of storing arrays for a small perf bump in the generated reducer:

```ts
if (map.has(type)) {
  const reducer = map.get(type) as ActionReducer<S, A>;
  map.set(type, (state, action) => on.reducer(reducer(state, action), action);
}
else {
  map.set(type, on.reducer);
}
```

Reply:
Thanks @MikeRyanDev for the great suggestion. I have pushed a commit. :)